fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergUtils.java

```diff
@@ -592,22 +592,17 @@ public static List<Column> getSchema(ExternalCatalog catalog, String dbName, Str
      * @return estimated row count
      */
     public static long getIcebergRowCount(ExternalCatalog catalog, String dbName, String tbName) {
-        try {
-            Table icebergTable = Env.getCurrentEnv()
-                    .getExtMetaCacheMgr()
-                    .getIcebergMetadataCache()
-                    .getIcebergTable(catalog, dbName, tbName);
-            Snapshot snapshot = icebergTable.currentSnapshot();
-            if (snapshot == null) {
-                // empty table
-                return 0;
-            }
-            Map<String, String> summary = snapshot.summary();
-            return Long.parseLong(summary.get(TOTAL_RECORDS)) - Long.parseLong(summary.get(TOTAL_POSITION_DELETES));
-        } catch (Exception e) {
-            LOG.warn("Fail to collect row count for db {} table {}", dbName, tbName, e);
+        Table icebergTable = Env.getCurrentEnv()
+                .getExtMetaCacheMgr()
+                .getIcebergMetadataCache()
+                .getIcebergTable(catalog, dbName, tbName);
+        Snapshot snapshot = icebergTable.currentSnapshot();
```

Comment:
icebergTable could be null?

Reply:
I checked the code, the table may be null when the iceberg metadata cache is not loaded. But I think it's not a problem, because the NPE would be caught in the caller and return the default value -1. Meanwhile it will trigger iceberg metadata cache to load the table so we can get it next time. I think we can fix this in a separate pr if needed. But I feel we don't need to do anything about it right now.

Reply:
add this explanation to comment?

Reply:
added